File: #1-25 (C, Java, Python)/Python/Euler7.py
#Euler7 ~ 10001st prime
import timeit
start = timeit.default_timer()

#Very unoptimized implementation of primality test
def is_prime_number(x):
    if x >= 2:
        for y in range(2,x):
            if not ( x % y ):
                return False
    else:
        return False
    return True

p = 0
i = 2
while(1):
    if is_prime_number(i):
        p = p + 1
        if p == 10001:
            break
    i = i + 1

stop = timeit.default_timer()
print("Solution: ",i,"\n")
print("Program completed in","{0:.3f}".format(stop - start), "seconds.")

File: #1-25 (C, Java, Python)/Java/Euler7.java
//Euler7 ~ 10001st prime
import java.util.concurrent.TimeUnit;

public class Euler7
{
    public static void main(String[] args) throws InterruptedException
	{
		long startTime = System.nanoTime();
		
		int  p = 0;
		long i = 2;
		do
		{
			if(primeCheck(i))
			{
				p++;
				if(p==10001)
				{
					break;
				}
			}
			i++;
		} while(true);
		
		long endTime = System.nanoTime();
		long timeElapsed = endTime - startTime;
		System.out.println("Solution: " + i + "\n");
		System.out.println("Program completed in " + timeElapsed/1000000000 + " seconds.");
    }
	
	//Very unoptimized implementation of primality test
	static boolean primeCheck(long n)
	{	
		for(long j = 2; j <= (n-1); j++)
		{
			if((n%j)==0)
			{
				return false;
			}
		}
		return true;
	}
}